### PR TITLE
Fixed a bug in non-English environments where the main hand was considered to be holding an item even though it was not holding an item

### PR DESCRIPTION
**Pull request by TheWhiteDog9487**
In the current version, if the Minecraft client is in a non-English language, when not holding any items, it will be misjudged because `client.player.getStackInHand(Hand.MAIN_HAND).getName().getString()` gets the name of the corresponding language instead of the English "Air", which actually displays "Holding Air".

<details>
    <summary>Screenshot</summary>
    Snipaste_2024-11-05_07-25-53
</details>

### CODE REVIEW

**Comment by ThePhaseCat** [comment]
If you can do it, that would be cool and would allow for other languages to be supported. I never actually knew this was a problem, so if it's possible to do it, that would be great, but I understand if it ends up not being possible!

**Comment by TheWhiteDog9487** [comment]
> If you can do it, that would be cool and would allow for other languages to be supported. I never actually knew this was a problem, so if it's possible to do it, that would be great, but I understand if it ends up not being possible!

OK, I will try my best to see if I can complete it, but I don't know if it will be completed in the end, and how long it will take.
Now if you think it's okay, you may merge or close this PR. If I complete this feature in the end, I will open a new pull request separately.

**Comment by TheWhiteDog9487** [comment]
@ThePhaseCat 
I have already completed this feature. The changes are huge. I tested it a bit and found no serious problems.
Should I push it directly to Girhub or open a new pull request after you merge it?
In addition, there is a problem I can't solve. If you switch the language in the game, the default values ​​of the fields that use translations, such as `mainAdvancedModeDetail` in the Mod Configuration interface, will not be changed. You must restart the game to change to the changed language. I don't know how to solve this.

**Comment by ThePhaseCat** [comment]
Could you make a separate pull request for that?

**Comment by TheWhiteDog9487** [comment]
> Could you make a separate pull request for that? 

If I push to this branch now, the two commits will be merged into this pull request. So I need to wait for this pull request to be merged, and the next time I open a pull request, there will only be one commit.

**Comment by TheWhiteDog9487** [comment]
> Could you make a separate pull request for that? 

If you don't want to merge this pull request now, then I need to push it to another branch, and you will need to merge it into the other branch later.

**Comment by ThePhaseCat** [comment]
If you give me like 5-6 hours I'll be able to merge this one if you're willing to wait

**Review by ThePhaseCat — APPROVED** [review]

**Comment by ThePhaseCat** [comment]
had time to merge in between classes!

**Comment by TheWhiteDog9487** [comment]
> had time to merge in between classes!

Thanks for help！

**Pull request merged**
